saleor/discount/forms.py

```diff
@@ -38,7 +39,7 @@ def clean(self):
                 discount = voucher.get_discount_for_checkout(self.checkout)
                 cleaned_data['discount'] = discount
             except NotApplicable as e:
-                self.add_error('voucher', unicode(e))
+                self.add_error('voucher', six.text_type(e))
```

Comment:
Could you use smart_text from django.utils.encoding? That's what we use in rest of the project.